### PR TITLE
Fixes slack verification and timeout errors

### LINKED ISSUE #2970: Slack Verification + Request Timeout
### Issue Summary
Currently the slack app fails verification on any "Interaction payloads" and times out if we don't response under 3 seconds

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?

Fixes slack timeout errors and verification failures and provides a more detailed error message to the user 

Fixes #2970 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/app-store/slackmessaging/api/add.ts [hunk 1]
```diff
@@ -14,10 +14,6 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   }
 
   if (req.method === "GET") {
-    if (!req.session?.user?.id) {
```
Comment (sean-brydon) [inline packages/app-store/slackmessaging/api/add.ts:17]: Duplicate code - check is  a few lines above

File: packages/app-store/slackmessaging/api/commandHandler.ts [hunk 2]
```diff
@@ -13,7 +13,6 @@ export enum SlackAppCommands {
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
   if (req.method === "POST") {
     const command = req.body.command.split("/").pop();
-    await slackVerify(req, res);
```
Comment (sean-brydon) [inline packages/app-store/slackmessaging/api/commandHandler.ts:16]: Slack dousing expect you to verify on commands - hence the verification failing

File: packages/app-store/slackmessaging/lib/showCreateEventMessage.ts [hunk 3]
```diff
@@ -35,5 +37,5 @@ export default async function showCreateEventMessage(req: NextApiRequest, res: N
     trigger_id: body.trigger_id,
     view: CreateEventModal(data),
   });
-  res.status(200).end();
+  return res.status(200).end();
```
Comment (sean-brydon) [inline packages/app-store/slackmessaging/lib/showCreateEventMessage.ts:40]: Was never returning here - Hence the timeout .... :(

File: packages/app-store/slackmessaging/lib/actions/createEvent.ts [hunk 4]
```diff
@@ -90,9 +93,7 @@ export default async function createEvent(req: NextApiRequest, res: NextApiRespo
       },
     });
 
-  const slackCredentials = foundUser?.credentials[0].key; // Only one slack credential for user
-
-  // @ts-ignore access_token must exist on slackCredentials otherwise we have wouldnt have reached this endpoint
+  const slackCredentials = foundUser?.credentials[0].key as { access_token: string }; // Only one slack credential for user
```
Comment (zomars) [inline packages/app-store/slackmessaging/lib/actions/createEvent.ts:96]: Use zod to verify this the case instead of casting

**Review by zomars — COMMENTED** [review]
LGMT. Left just one comment

**Review by zomars — APPROVED** [review]
Made some improvements. Feel free to retest and merge

**Pull request merged**